### PR TITLE
Bug fix to fabric and ethereum connector

### PR DESCRIPTION
**Pull request by Ram-srini**
1. Submit work order only if active worker id exists in
    the work order request from blockchain event.
 2. Condition to handle synchronous work order processing response

Signed-off-by: Ramakrishna Srinivasamurthy <ramakrishna.srinivasamurthy@intel.com>

### CODE REVIEW

File: blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py [hunk 1]
```diff
@@ -122,50 +124,57 @@ def workorder_event_handler_func(self, event, block_num, txn_id, status):
...
-            # get the work order result
-            jrpc_req_id += 1
-            work_order_result = \
-                self.__jrpc_work_order.work_order_get_result(
-                    work_order_req['workOrderId'],
-                    jrpc_req_id
-                )
-            logging.info("Work order get result {}".format(
-                work_order_result
+        # Submit the work order to listener if worker id the event
```
Comment (danintel) [inline blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py:127]: Change "worker id the event" to "worker id from the event"

File: blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py [hunk 2]
```diff
@@ -122,50 +124,57 @@ def workorder_event_handler_func(self, event, block_num, txn_id, status):
...
+        # matches with available worker ids
+        if work_order_req['workerId'] in self.__worker_ids:
```
Comment (manojgop) [inline blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py:129]: 1) Is this issue applicable for ethereum as well ? Then we need to do similar change in ethereum.

2) If worker id in request does match available worker ids in Avalon KV storage can we return error.

File: blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py [hunk 3]
```diff
@@ -122,50 +124,57 @@ def workorder_event_handler_func(self, event, block_num, txn_id, status):
...
+                        work_order_req['workOrderId'],
+                        jrpc_req_id
+                    )
+                logging.info("Work order get result {}".format(
+                    work_order_result
+                ))
+            # With ZMQ based work order handler in listener returns result
+            # as a response for work order submit
+            elif response and 'result' in response:
+                work_order_result = response
```
Comment (manojgop) [inline blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py:157]: Please add similar change for ethereum. We could rephrase the comment "ZMQ based work order handler " to " With Synchronous work order processing" . ZMQ could be replaced with other approach for synchronous work order processing later.

Comment (Ram-srini) [reply blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py:157]: Currently ethereum is not validating work order submit response. I can add similar work order submit response validation in ethereum as well.

File: blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py (hunk 2 above)
Comment (Ram-srini) [reply blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py:129]: 1. Same changes required in thereum and i will do it.
2. check is there in https://github.com/hyperledger/avalon/blob/master/listener/avalon_listener/tcs_work_order_handler.py#L121
not in tcs_work_order_handler_sync.py we can similar check here also.

File: blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py (hunk 1 above)
Comment (Ram-srini) [reply blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py]: Done

**Pull request merged**